File: algorithm/src/main/java/baekjoon/p11575/Main.java
package baekjoon.p11575;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.nextInt(), sc.nextInt(), sc.next()));
        }
    }

    private static String solve(int a, int b, String str) {
        StringBuilder sb = new StringBuilder();

        for (char ch : str.toCharArray()) {
            int x = ch - 'A';
            int n = (a * x + b) % 26;
            sb.append((char) (n + 'A'));
        }

        return sb.toString();
    }

}

File: algorithm/src/main/java/baekjoon/p11575/Main.kt
package baekjoon.p11575

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        val (a, b) = readLine()!!.split(" ").map { it.toInt() }
        val str = readLine()!!
        println(solve(a, b, str))
    }
}

private fun solve(a: Int, b: Int, str: String): String {
    return buildString {
        for (ch in str) {
            val x = ch - 'A'
            val n = (a * x + b) % 26
            val ret = (n + 'A'.toInt()).toChar()
            append(ret)
        }
    }
}